### ISSUE TITLE
Android build script almost wiped off my OS!

### ISSUE THREAD

**Issue by kakyoism** [zeromq/libzmq#3979]
*Please use this template for reporting suspected bugs or requests for help.*

# Issue description

if defined `ANDROID_BUILD_CLEAN` environment variable without defining ANDROID_BUILD_PREFIX, 

we get an `rm -rf /*`

# Environment

* libzmq version (commit hash if unreleased): 4.3.2
* OS: macOS

# Minimal test code / Steps to reproduce the issue

```sh
Host=darwin-x86_64
export ANDROID_BUILD_CLEAN=1
export TOOLCHAIN_NAME=llvm
export TOOLCHAIN_PREFIX="${ANDROID_NDK_ROOT}/toolchains/${TOOLCHAIN_NAME}/prebuilt/$Host"
export TOOLCHAIN_HOST=aarch64-linux-android
export TOOLCHAIN_PATH="${TOOLCHAIN_PREFIX}/bin"
export TOOLCHAIN_ARCH=arm64
export TOOLCHAIN_LEVEL=26
export ANDROID_BUILD_CXXSTL=gnustl_shared_49
```

Then run `zeromq/builds/android/build.sh`

# What's the actual result? (include assertion message & call stack if applicable)

The following piece of code in `build.sh` becomes `rm -rf /*`!!!

```sh
# Check for environment variable to clear the prefix and do a clean build
if [[ $ANDROID_BUILD_CLEAN ]]; then
    echo "Doing a clean build (removing previous build and depedencies)..."
    rm -rf "${ANDROID_BUILD_PREFIX}"/*
fi
```

# What's the expected result?

Build script should be safe and fool-proof in terms of wiping off the whole system!!!

**Comment by kakyoism** [comment]
Well. Just realized that I removed the initialization code of ANDROID_BUILD_PREFIX. So it was actually my own fault. But all in all, `rm -rf` shouldn't be used with wildcard. And we should really `cd` into a sub folder for any `rm`.

**Comment by bluca** [comment]
can you send a PR?

**Comment by benjdero** [comment]
ANDROID_BUILD_PREFIX is set by `android_build_set_env` function:
```bash
export ANDROID_BUILD_PREFIX="${ANDROID_BUILD_DIR}/prefix/${TOOLCHAIN_ARCH}"
```

How can it be null at this point?

**Comment by kakyoism** [comment]
> ANDROID_BUILD_PREFIX is set by `android_build_set_env` function:
> 
> ```shell
> export ANDROID_BUILD_PREFIX="${ANDROID_BUILD_DIR}/prefix/${TOOLCHAIN_ARCH}"
> ```
> 
> How can it be null at this point?

Indeed, I removed that part just because I found no way to override prefix. So it's partly my problem. But the `rm -rf /*` is really dangerous. 

The current build script does not build on macOS because it's using an outdated Android toolchain. I had an ugly hack but I'd really love to have a proper fix to the build script.

See issue: https://github.com/zeromq/libzmq/issues/3764
